arwen/host/asyncCall.go

```diff
@@ -51,21 +51,72 @@ func (host *vmHost) handleAsyncCallBreakpoint() error {
 	return nil
 }
 
-func (host *vmHost) canExecuteSynchronouslyOnDest(dest []byte) bool {
+func (host *vmHost) determineAsyncCallExecutionMode(asyncCallInfo *arwen.AsyncCallInfo) (arwen.AsyncCallExecutionMode, error) {
+	runtime := host.Runtime()
 	blockchain := host.Blockchain()
-	calledSCCode, err := blockchain.GetCode(dest)
 
-	return len(calledSCCode) != 0 && err == nil
+	// If ArgParser cannot read the Data field, then this is neither a SC call,
+	// nor a built-in function call.
+	argParser := parsers.NewCallArgsParser()
+	functionName, _, err := argParser.ParseData(string(asyncCallInfo.Data))
+	if err != nil {
+		return arwen.AsyncUnknown, err
+	}
+
+	code, err := blockchain.GetCode(asyncCallInfo.Destination)
+	if len(code) > 0 && err == nil {
```

Comment:
If there's an `err` here we continue, but completely discard the error. Correct?

Reply:
yes. correct.